libraries/cbor/src/writer.rs

```diff
@@ -54,37 +74,29 @@ impl<'a> Writer<'a> {
             Value::Array(array) => {
                 self.start_item(type_label, array.len() as u64);
                 for el in array {
-                    if !self.encode_cbor(el, remaining_depth - 1) {
-                        return false;
-                    }
+                    self.encode_cbor(el, remaining_depth.map(|d| d - 1))?;
                 }
             }
             Value::Map(mut map) => {
                 map.sort_by(|a, b| a.0.cmp(&b.0));
                 let map_len = map.len();
                 map.dedup_by(|a, b| a.0.eq(&b.0));
                 if map_len != map.len() {
-                    return false;
+                    return Err(EncoderError::DuplicateMapKey);
                 }
                 self.start_item(type_label, map_len as u64);
                 for (k, v) in map {
-                    if !self.encode_cbor(k, remaining_depth - 1) {
-                        return false;
-                    }
-                    if !self.encode_cbor(v, remaining_depth - 1) {
-                        return false;
-                    }
+                    self.encode_cbor(k, remaining_depth.map(|d| d - 1))?;
+                    self.encode_cbor(v, remaining_depth.map(|d| d - 1))?;
                 }
             }
             Value::Tag(tag, inner_value) => {
                 self.start_item(type_label, tag);
-                if !self.encode_cbor(*inner_value, remaining_depth - 1) {
-                    return false;
-                }
+                self.encode_cbor(*inner_value, remaining_depth.map(|d| d - 1))?;
```

Comment:
I took a quick look at the CTAP2 spec and its nesting limits are specifically for arrays and maps.  Should a tagged value should count towards the nesting limits?  (i.e. should this line be just `self.encode_cbor(*inner_value, remaining_depth)?;` ?)

Reply:
CTAP2 does not allow tags, so I'd assume it's simply not mentioned. We should ignore tags whenever possible in CTAP2.

I did not find an explicit answer to whether tags reduce the nesting depth or not in the RFC. Other libraries do reduce, so I think we should be fine.